modules/tinymce/src/plugins/image/main/ts/ui/Dialog.ts

```diff
@@ -415,6 +407,8 @@ const parseStyle = (editor: Editor) => (cssText: string): StyleMap => editor.dom
 
 const serializeStyle = (editor: Editor) => (stylesArg: StyleMap, name?: string): string => editor.dom.serializeStyle(stylesArg, name);
 
+const uploadImage = (editor: Editor) => (blobInfo: BlobInfo) => ImageUploader(editor).upload([ blobInfo ], false).then((results) => results[0]);
```

Comment:
Is it possible for an empty array to return? At a glance it looks like it might be if a custom upload handler is provided?

Reply:
Might be worth checking what happens when there is a failure while uploading the image

Reply:
@JamesToohey  If I'm not wrong is all handle inside Uploader.ts already.

Reply:
Done @JamesToohey and @leemhenson , just pushed the changes to address this scenario.